vendor/ddcloud/resource_additional_nics.go

```diff
@@ -55,21 +71,314 @@ func resourceAdditionalNic() *schema.Resource {
 }
 
 func resourceAdditionalNicCreate(data *schema.ResourceData, provider interface{}) error {
-	var err error
-	return err
+	//propertyHelper := propertyHelper(data)
+	apiClient := provider.(*providerState).Client()
+	serverID := data.Get(resourceKeyNicServerID).(string)
+	ipv4Address := data.Get(resourceKeyNicPrivateIPV4).(string)
+	vlanID := data.Get(resourceKeyNicVLANID).(string)
+
+	shutdownOk := data.Get(resourceKeyNicIsShutdownOk).(bool)
+
+	log.Printf("Configure additional nics for server '%s'...", serverID)
+
+	server, err := apiClient.GetServer(serverID)
+	if err != nil {
+		return err
+	}
+	isStarted := server.Started
+	if isStarted {
```

Comment:
if the shutdown_ok option is false and the server is powered on, then the server will not be shut down and add nic api call will throw the error when trying to add the nic. If the server is powered off and the shutdown_ok option is false, then Add Nic Operation will be successful.